tests/e2e/test_rosacli_node_pool.go

```diff
@@ -730,4 +730,78 @@ var _ = Describe("Edit nodepool",
+				Expect(err).ToNot(HaveOccurred())
+				Expect(help.String()).To(ContainSubstring("--node-drain-grace-period"))
+				help, err = machinePoolService.RetrieveHelpForEdit()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(help.String()).To(ContainSubstring("--node-drain-grace-period"))
+
+				nodeDrainGracePeriodsReqAndRes := []map[string]string{{"20": "20 minutes", "20 hours": "1200 minutes", "20 minutes": "20 minutes"}}
+
+				for _, nodnodeDrainGracePeriod := range nodeDrainGracePeriodsReqAndRes {
+					for req, res := range nodnodeDrainGracePeriod {
+
+						nodePoolName := common.GenerateRandomName("np-72715", 2)
+						By("Create nodepool with node-drain-grace-period")
+						_, err = machinePoolService.CreateMachinePool(clusterID, nodePoolName,
+							"--replicas", "3",
+							"--node-drain-grace-period", req,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Describe nodepool")
+						output, err := machinePoolService.DescribeAndReflectNodePool(clusterID, nodePoolName)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(output.NodeDrainGracePeriod).To(Equal(res))
+					}
+				}
+				nodePoolName := common.GenerateRandomName("np-72715", 3)
+				By("Create nodepool without node-drain-grace-period")
+				_, err = machinePoolService.CreateMachinePool(clusterID, nodePoolName,
+					"--replicas", "3",
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Describe cluster in json format")
+				rosaClient.Runner.JsonFormat()
+				jsonOutput, err := clusterService.DescribeCluster(clusterID)
+				Expect(err).To(BeNil())
+				rosaClient.Runner.UnsetFormat()
+				jsonData := rosaClient.Parser.JsonData.Input(jsonOutput).Parse()
+				value := jsonData.DigFloat("node_drain_grace_period", "value")
+				nodeDrainGracePeriodForCluster := strconv.FormatFloat(value, 'f', -1, 64)
+
+				By("Describe nodepool")
+				output, err := machinePoolService.DescribeAndReflectNodePool(clusterID, nodePoolName)
+				Expect(err).ToNot(HaveOccurred())
+				if nodeDrainGracePeriodForCluster == "0" {
+					Expect(output.NodeDrainGracePeriod).To(Equal(""))
+				} else {
+					Expect(output.NodeDrainGracePeriod).To(Equal(nodeDrainGracePeriodForCluster))
+				}
+
+				nodeDrainGracePeriodsReqAndRes = []map[string]string{{"10": "10 minutes", "10 hours": "600 minutes", "10 minutes": "10 minutes"}}
+				for _, nodnodeDrainGracePeriod := range nodeDrainGracePeriodsReqAndRes {
+					for req, res := range nodnodeDrainGracePeriod {
+
+						By("Edit nodepool with node-drain-grace-period")
+						_, err = machinePoolService.EditMachinePool(clusterID, nodePoolName,
+							"--node-drain-grace-period", req,
+							"--replicas", "3",
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Describe nodepool")
+						output, err := machinePoolService.DescribeAndReflectNodePool(clusterID, nodePoolName)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(output.NodeDrainGracePeriod).To(Equal(res))
+					}
+				}
+			})
```

Comment:
We'd better to do clean-up step after the testing, in this case, to delete the created machinepools in this case

Reply:
It is being cleaned through afterEach() function inside describe block, so NO need to duplicate it I guess, wdyt?

Reply:
It is fine if there is clean-up step in afterEach block.

Reply:
Thanks